### DISCUSSION TITLE
Lost Power Meter Consumption and Exported after installing battery

### DISCUSSION THREAD

**Discussion by webdados** [wlcrs/huawei_solar#721] [Setting up the integration]
Hi,

I installed a battery 2 days ago and now realized I don't have Consumption and Exported data on the Energy dashboard:
<img width="1135" alt="image" src="https://github.com/wlcrs/huawei_solar/assets/6853605/f8641fad-010a-4a52-882c-d6571b3703c3">

This is due to these sensors now being "Unknown":
<img width="678" alt="image" src="https://github.com/wlcrs/huawei_solar/assets/6853605/0144c386-0f58-4e70-8958-a0ac18b5b39a">

Any ideas? I already restarted the integration and Home Assistant without luck.

**Comment by Roving-Ronin** [comment]
@webdados strange as first thought is, is the RS485 correctly wired up, but it still has Active Power / PF showing. You should probably check if consumption/export are being updated in FusionSolar portal, and if they are then you know its an issue in the HA side only.

If its HA, then delete the plant from HA and re-add, and/or delete WCLRS, reboot, and then reinstall and reboot.
If it stil occurs and/or isn't HA (ie not showing up in FusionSolar portal) then I'd be inclined to turn off the PV input, turn the inverters/battery all of at the breakers, then turn off the grid feed to the house. This should 'turn off' the DDSU-666H meter and allow 'rebooting' it when you power it all back on again. Is that doesn't work I'd be looking up if there is a procedure to reset the meter.

**Reply by webdados** [reply]
Everything seems ok in FusionSolar:
![Screenshot 2024-06-03 at 13 41 25](https://github.com/wlcrs/huawei_solar/assets/6853605/acd8d2b3-5252-44ad-a18b-346648a6f329)

I've updated WLCRS to the last version and restarted HA and it's not fixed.

If I remove WLCRS how can I guarantee that when I add it again, all the sensors have the same id as they have now?

**Reply by Roving-Ronin** [reply]
@webdados did you start in v1.3.3 as there was a change in entity ids going to that version, can't remember specifics off hand, but it was along the lines of naming changed for multiple devices where say sensor.inverter2_activepower became sensor.inverter_activepower_2
etc.

I've recently had to delete the ola t, reinstall the integration and it came back up with the same entity names each time.
At worst if you have a couple you've renamed, you'd find the new names and also see the new named versions of them... then just copy the new entity name, delete the entity and then rename your old entity to the new naming standard and your old entities will have the new names, will be working and tied back to the old data stored by HA Recorder.

**Reply by webdados** [reply]
How do you recommend to reinstall the integration? Remove in HACS, restart, add again in HACS?

**Reply by Roving-Ronin** [reply]
To start with just delete the plant in HA, then re-add making sure of the Modbus id numbers and order of primary inverter id# then secondary inverter.

**Reply by webdados** [reply]
Hi,

The "plant"? How do I remove the plant?
Shouldn't I simply remove the all integration?

Where in the config files can I find the modbus, connection mode, ips, etc?

Is this it on core.config_entries?

```
{
        "entry_id": "2db972f6ccb13cf25d627c6e7eb70e37",
        "version": 1,
        "minor_version": 1,
        "domain": "huawei_solar",
        "title": "SUN2000-6KTL-L1",
        "data": {
          "host": "192.168.1.244",
          "port": 502,
          "slave_ids": [
            1
          ],
          "enable_parameter_configuration": true,
          "username": null,
          "password": null
        },
        "options": {},
        "pref_disable_new_entities": false,
        "pref_disable_polling": false,
        "source": "user",
        "unique_id": "102150268845",
        "disabled_by": null
}
```

**Comment by webdados** [comment]
Hi @Roving-Ronin 

How can I "delete the plant" in HA?

**Comment by webdados** [comment]
Anyway, just rechecked and I don't have consumption on FusionSolar either:
<img width="646" alt="image" src="https://github.com/user-attachments/assets/cd39cc75-c208-4586-9cb9-92b8a4219d34" />

**Comment by webdados** [comment]
Any ideas on why this could happen? Seems to be a problem on the reading from the power meter.
